File: Single-Linked-List/delete_end.py
# File Name: delete_end.py
# Program to delete the last node of a singly linked list

class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

head = None

# Function to delete the last node
def delete_end():
    global head

    if head is None:
        print("List is empty. Deletion not possible.")
        return

    # If only one node exists
    if head.next is None:
        head = None
        print("Last node deleted successfully.")
        return

    temp = head
    while temp.next.next is not None:
        temp = temp.next
    temp.next = None
    print("Last node deleted successfully.")

if __name__ == "__main__":
    delete_end()

File: Single-Linked-List/DeleteEnd.java
// File Name: DeleteEnd.java
// Program to delete the last node of a singly linked list

class Node {
    int data;
    Node next;

    Node(int data) {
        this.data = data;
        this.next = null;
    }
}

public class DeleteEnd {
    static Node head = null;

    // Function to delete the last node
    static void delete_end() {
        if (head == null) {
            System.out.println("List is empty. Deletion not possible.");
            return;
        }

        // If only one node exists
        if (head.next == null) {
            head = null;
            System.out.println("Last node deleted successfully.");
            return;
        }

        Node temp = head;
        while (temp.next.next != null) {
            temp = temp.next;
        }
        temp.next = null;
        System.out.println("Last node deleted successfully.");
    }

    public static void main(String[] args) {
        delete_end();
    }
}
